keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+                cpu: 2.5
+            containers:
+            - name: container-1
+                resources:
+                  requests:
+                    memory: 60Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 300Mi
+                    cpu: 2  
+            - name: container-2
+                resources:
+                  requests:
+                    memory: 40Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 200Mi
+                    cpu: 1 
+          ```
+
+          The pod-level limit will serve as an aggregate ceiling. This will cover the scenario where the sum of peak usage for individual containers in a pod is high, but these
+          peaks don’t necessarily occur simultaneously.
+
+          If we simply set the pod-level limit as the sum of the individual container limits, we might overallocate resources. The pod might never actually use all those resources
+          at once, leading to wasted capacity on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+
+#### Scheduler Changes
+
+The scheduler logic needs to be updated to support pod-level requests and limits.
+
+* **Scenario 1: Only container-level resources are specified**
+
+  No changes, same as existing.
+
+* **Scenario 2: Only pod-level resources are specified**
+
+  Consider pod-level requests to find the suitable node for pod placement.
+
+* **Scenario 3: Both container-level and pod-level resources are specified**
+
+  Scheduler will prioritize pod-level requests as the sole indicator of the pod’s resources needs. It will ignore container level requests. This ensures that the pod 
+  is scheduled on a node that can fulfill the overall resource requirements of the pod, regardless of the individual container requests.
+
+
+#### QoS Changes
+
+The QoS definitions need to be updated to consider pod-level resources. To attach a QoS class to a pod, the  pod’s resource specifications are evaluated in order:
+
+1. The values for resources specified in the pod’s `resources` stanza.
+2. The value of resources specified in the container’s `resources` stanza.
+
+In simple words, if resources are specified at pod-level, the values in pod-level `resources stanza` are given a preference in determining the QoS class.
+
+* **Guaranteed QoS Criteria**
+
+    One of the following two conditions must apply:
+
+    * Pod level requests and limits are specified for each resource, and request is equal to limit for each resource. The container level resource specification 
+    (specified or not specified) is not taken into account for QoS class determination in this case.
+    * In absence of pod-level resources specifications, every container in the pod has requests and limits specified for each resource and all container requests
```

Comment:
in absence - does it mean both needs to be omited? If pod has limit defined, but not requests, than it is NOT guaranteed, correct?

Reply:
Correct, Reframed this section.